### PR TITLE
reduce getHeaders() call in Request::setServer()

### PR DESCRIPTION
**Pull request by samsonasik**
- [x] Is this related to quality assurance?

### CODE REVIEW

File: src/PhpEnvironment/Request.php [hunk 1]
```diff
@@ -233,7 +233,7 @@ public function setServer(ParametersInterface $server)
             }
         }
 
-        $this->getHeaders()->addHeaders($headers);
+        $headers = $this->getHeaders()->addHeaders($headers);
```
Comment (webimpress) [inline src/PhpEnvironment/Request.php]: Personally I am not sure if it is good improvement, I mean, for me it is less readable - as definition of the variable and usage of it is several lines below.
Also it takes advantages of fluent interfaces, and as far as I know we tend to drop them.

File: src/PhpEnvironment/Request.php [hunk 2]
```diff
@@ -266,8 +266,8 @@ public function setServer(ParametersInterface $server)
         $port = null;
 
         // Set the host
-        if ($this->getHeaders()->get('host')) {
-            $host = $this->getHeaders()->get('host')->getFieldValue();
+        if ($headers->get('host')) {
```
Comment (webimpress) [inline src/PhpEnvironment/Request.php]: We have then twice `->get('host')`.
What about:
```php
$headerHost = $this->getHeaders()->get('host');
if ($headerHost) {
    $host = $headerHost->getFieldValue();
    ...
```

File: src/PhpEnvironment/Request.php (hunk 1 above)
Comment (samsonasik) [reply src/PhpEnvironment/Request.php]: Ok, I reverted the change of re-assignment headers.

File: src/PhpEnvironment/Request.php (hunk 2 above)
Comment (samsonasik) [reply src/PhpEnvironment/Request.php]: Ok, implemented assign first to `$headerHost`

**Comment by webimpress** [comment]
Thanks, @samsonasik!

**Pull request closed without merging**